cloudpickle/cloudpickle.py

```diff
@@ -934,7 +939,10 @@ def subimport(name):
 def dynamic_subimport(name, vars):
     mod = imp.new_module(name)
     mod.__dict__.update(vars)
-    sys.modules[name] = mod
+    # adding a dynamic module to sys.module can cause recursive imports
+    # in python 2.x.x because site.Quitter contains sys as one of its globals
+    if sys.version > '3':
+        sys.modules[name] = mod
```

Comment:
Mutating `sys.modules` only under Python 3 seems weird to me. Why the discrepancy between the two major versions? Why is this not required for Python 2?